src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -2017,6 +2000,18 @@ static void on_connection_event_CSIDE_IO(pn_event_t *e, qd_server_t *qd_server,
     }
 
     connection_run_CSIDE_IO(conn);
+
+    qd_tcp_listener_t *connector = (qd_tcp_listener_t*) conn->common.parent;
+    sys_mutex_lock(&connector->lock);
+    if(conn->management_close) {
```

Comment:
Could we make the `conn->management_close` flag atomic? Then we could avoid taking the `connector->lock` here (for every event of every connection).